server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/dto/WebAdminSingleMailboxReindexingTaskAdditionalInformationDTO.java

```diff
@@ -0,0 +1,94 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.webadmin.dto;
+
+import java.time.Instant;
+
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.james.json.DTOModule;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTO;
+import org.apache.james.server.task.json.dto.AdditionalInformationDTOModule;
+import org.apache.mailbox.tools.indexer.SingleMailboxReindexingTask;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class WebAdminSingleMailboxReindexingTaskAdditionalInformationDTO implements AdditionalInformationDTO {
+
+    public static AdditionalInformationDTOModule<SingleMailboxReindexingTask.AdditionalInformation, WebAdminSingleMailboxReindexingTaskAdditionalInformationDTO> serializationModule(MailboxId.Factory factory) {
+        return DTOModule.forDomainObject(SingleMailboxReindexingTask.AdditionalInformation.class)
+            .convertToDTO(WebAdminSingleMailboxReindexingTaskAdditionalInformationDTO.class)
+            .toDomainObjectConverter(dto -> {
+                throw new NotImplementedException("Deserialization not implemented for this DTO");
+            })
+            .toDTOConverter((details, type) -> new WebAdminSingleMailboxReindexingTaskAdditionalInformationDTO(
+                type,
+                details.getMailboxId(),
+                details.getSuccessfullyReprocessedMailCount(),
+                details.getFailedReprocessedMailCount(),
+                SerializableReIndexingExecutionFailures.from(details.failures()),
+                details.timestamp()))
+            .typeName(SingleMailboxReindexingTask.MAILBOX_RE_INDEXING.asString())
```

Comment:
`MAILBOX_RE_INDEXING` is confusing, can you use `TYPE` instead?

Reply:
I don't understand. Can you be clearer?

Reply:
usually (there are only a handful of exceptions), the constant representing a `Task`'s `Type` is named `TYPE`, not the `Task`'s name

Reply:
IMO you are asking for too low level changes unrelated to my code.

Will do but in the future I would really like that we split concerns. Sharing the burdon of these refactorings would also be welcome.